Shipyard integration

mesmith027 opened the pull request:
### What does this PR do?

Creating a standard API integration PR for (Shipyard)[https://shipyard.build/]

### Motivation

Shipyard is in the process of partnering with Datadog. At the instructions of Dominic Medina, I am working through the different stages of the partnership. This is the second of 3 steps. 

### Review checklist

- [X] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [X] Git history is clean
- [N/A] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

I was unsure of how to reference the picture that should be displayed in the README from `images/datadog-input.png`, so please let me know if I need to make changes to this link to have it display properly.

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+
+## Overview
+
+The Datadog Shipyard integration passes your [Shipyard](1) ephemeral environment logs to your Datadog dashboard. This is useful for using Datadog's [log analysis and correlation](https://www.datadoghq.com/solutions/log-analysis-and-correlation/) functions and helping your team with [shift left testing](https://www.datadoghq.com/solutions/shift-left-testing/).
```
```suggestion
The Datadog Shipyard integration passes your [Shipyard](1) ephemeral environment logs to your Datadog dashboard. This is useful for using Datadog's [log analysis and correlation][8] functions and helping your team with [shift left testing][9].
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+[4]: https://docs.datadoghq.com/getting_started/site/#access-the-datadog-site
+[5]: https://app.datadoghq.com/organization-settings/api-keys
+[6]: https://raw.githubusercontent.com/mesmith027/DataDog-integrations-extras/shipyard-integration/shipyard/images/datadog-input.png
+[7]: https://docs.shipyard.build/docs/integrations/#send-logs-to-datadog
```
```suggestion
[7]: https://docs.shipyard.build/docs/integrations/#send-logs-to-datadog
[8]: https://www.datadoghq.com/solutions/log-analysis-and-correlation/
[9]: https://www.datadoghq.com/solutions/shift-left-testing/
[10]: https://docs.shipyard.build/docs/config
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+
+## Setup
+
+These instructions assume you have a Shipyard account and application set up. If you do not, you can get started for [free here](2). You can also follow the [Shipyard docs](7) on how to set up the Datadog integration. 
```
```suggestion
These instructions assume you have a Shipyard account and application set up. If you do not, you can get started [for free][2]. For more information about setting up the Datadog integration, see the [Shipyard documentation][7]. 
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+
+### Configuration
+
+0. Go to your [Datadog APIs page][5] and create an API key.
```
```suggestion
1. Navigate to the [API Keys page][5] and create an API key.
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+### Configuration
+
+0. Go to your [Datadog APIs page][5] and create an API key.
+1. In your Shipyard dashboard, go to settings. 
```
```suggestion
2. In your Shipyard dashboard, go to **Settings**. 
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+1. In your Shipyard dashboard, go to settings. 
+2. Under the Datadog section, enter your Datadog API key and the Datadog site URL.
+   ![datadog-input][6]
+   1. If you do not see a Datadog section here, email [support@shipyard.build](3) to enable this feature for your organization.
```
```suggestion
   1. If you do not see a Datadog section here, email [support@shipyard.build][3] to enable this feature for your organization.
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+
+0. Go to your [Datadog APIs page][5] and create an API key.
+1. In your Shipyard dashboard, go to settings. 
+2. Under the Datadog section, enter your Datadog API key and the Datadog site URL.
```
```suggestion
3. Under the Datadog section, enter your Datadog API key and the Datadog site URL.
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+2. Under the Datadog section, enter your Datadog API key and the Datadog site URL.
+   ![datadog-input][6]
+   1. If you do not see a Datadog section here, email [support@shipyard.build](3) to enable this feature for your organization.
+   2. To get the Datadog site URL, use the [Getting Started with Datadog Sites](4) documentation and compare your dashboard's site URL to the ones in the table.
```
```suggestion
   2. To get the Datadog site URL, see [Getting Started with Datadog Sites][4] and compare your dashboard's site URL to the ones in the table.
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+   ![datadog-input][6]
+   1. If you do not see a Datadog section here, email [support@shipyard.build](3) to enable this feature for your organization.
+   2. To get the Datadog site URL, use the [Getting Started with Datadog Sites](4) documentation and compare your dashboard's site URL to the ones in the table.
+3. Click the "install Datadog" button.
```
```suggestion
3. Click **Install Datadog**.
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+   1. If you do not see a Datadog section here, email [support@shipyard.build](3) to enable this feature for your organization.
+   2. To get the Datadog site URL, use the [Getting Started with Datadog Sites](4) documentation and compare your dashboard's site URL to the ones in the table.
+3. Click the "install Datadog" button.
+   1. Optionally, toggle the "enable Datadog logging for all environments," Shipyard will automatically send the logs for all your environments.
```
```suggestion
   1. Optionally, toggle `enable Datadog logging for all environments`, Shipyard will automatically send the logs for all your environments.
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+   2. To get the Datadog site URL, use the [Getting Started with Datadog Sites](4) documentation and compare your dashboard's site URL to the ones in the table.
+3. Click the "install Datadog" button.
+   1. Optionally, toggle the "enable Datadog logging for all environments," Shipyard will automatically send the logs for all your environments.
+4. To select specific environments from which you want to send logs for each repo, visit the [Configure Application page](https://docs.shipyard.build/docs/config). 
```
```suggestion
4. To select specific environments from which you want to send logs for each repo, see the [Configure Application page][10]. 
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+3. Click the "install Datadog" button.
+   1. Optionally, toggle the "enable Datadog logging for all environments," Shipyard will automatically send the logs for all your environments.
+4. To select specific environments from which you want to send logs for each repo, visit the [Configure Application page](https://docs.shipyard.build/docs/config). 
+   1. Select the third tab, "Notification Settings". 
```
```suggestion
   1. Select the **Notification Settings** tab. 
```

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+   1. Optionally, toggle the "enable Datadog logging for all environments," Shipyard will automatically send the logs for all your environments.
+4. To select specific environments from which you want to send logs for each repo, visit the [Configure Application page](https://docs.shipyard.build/docs/config). 
+   1. Select the third tab, "Notification Settings". 
+   2. Toggle the "Enable Datadog logging" for this environment.
```
```suggestion
   2. Toggle `Enable Datadog logging` for this environment.
```

alai97 reviewed (changes requested):
Some copy suggestions!

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,49 @@
+0. Go to your [Datadog APIs page][5] and create an API key.
+1. In your Shipyard dashboard, go to settings. 
+2. Under the Datadog section, enter your Datadog API key and the Datadog site URL.
+   ![datadog-input][6]
```
This is correct! :D

emarsha94 commented on shipyard/README.md:
```diff
@@ -0,0 +1,52 @@
+
+## Overview
+
+The Datadog Shipyard integration passes your [Shipyard][1] ephemeral environment logs to your Datadog dashboard. This is useful for using Datadog's [log analysis and correlation][8] functions and helping your team with [shift-left testing][9].
+
```
The overview section should include:
1. What Shipyard does in 1-2 sentences
2. The value that customers would get from this integration 
3. Data you are sending in

mesmith027 replied:
I believe I have included the necessary details for you here: 0416cd420dbea24e7aebb1351f6ad906b3a7bd41

emarsha94 commented on shipyard/images/IMAGES_README.md:
This file can be removed

emarsha94 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,44 @@
+  "manifest_version": "2.0.0",
+  "app_uuid": "3ea4d492-9d1c-4dda-bb8a-e571eca3ce7d",
+  "app_id": "shipyard",
+  "display_on_public_website": false,
```
```suggestion
  "display_on_public_website": true,
```

emarsha94 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,44 @@
+    "media": [
+      {
+        "media_type": "image",
+        "caption": "",
```
Please provide a caption of the image

mesmith027 replied:
Added captions for both images here: e24ab66684f3948876dfdebb39202df0136969df

emarsha94 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,44 @@
+      },
+      {
+        "media_type": "image",
+        "caption": "",
```
Please provide a caption of the image

mesmith027 replied:
Added captions for both images here: e24ab66684f3948876dfdebb39202df0136969df

emarsha94 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,44 @@
+      }
+    ],
+    "classifier_tags": [
+      "Supported OS::Linux",
+      "Supported OS::Windows",
+      "Supported OS::macOS"
```
Please add relevant categories and submitted data from [here](https://docs.datadoghq.com/developers/integrations/check_references/#classifier-tags)

mesmith027 replied:
Thanks for the link I have added relevant classifier tags here: 9526ff0b2936103419beb37f9d845451ac936117

emarsha94 reviewed (changes requested):
Thanks for submitting the PR! Excited for the integration. I left some comments/suggestions within the PR. Additionally, you are missing the CODEOWNERS file. Please make sure to add your github handle to the codeowners file. For [reference](https://docs.datadoghq.com/developers/marketplace/offering#complete-the-necessary-integration-asset-files)

emarsha94 commented on shipyard/images/shipyard-teal.png:
Please provide a logo in .svg format

mesmith027 replied:
Addressed logo issues here: d59227906284e2a68ff54c01cc804629b14beefb

mesmith027 commented:
Hi @emarsha94, 

Thanks for your help and guidance on this. I have addressed all your comments and made the necessary edits, but a few checks still need to be fixed, and I am still trying to figure out how to fix them. 

Can you shed some light on what else needs fixing to finish this PR?

emarsha94 commented on shipyard/README.md:
```diff
@@ -0,0 +1,56 @@
+
+Shipyard unblocks DevOps teams from bulky, outdated production pipelines. We resolve the pre-production or staging bottleneck with our agile on-demand ephemeral (testing or preview) environments.
+
+The Datadog Shipyard integration automatically passes your [Shipyard][1] ephemeral environment logs to your Datadog dashboard. This is valuable for using Datadog's [log analysis and correlation][8] functions and helping your team with [shift-left testing][9].
+
+For every application you want the logs for, Shipyard sends the runtime logs of your application to Datadog. 
```
Please include what kinds of logs will be sent in, and the value that these logs will bring for a Datadog/Shipyard customer - i.e. why they want these logs, and how they can use them in Datadog. 

In this case, it links to various articles which could be a lot of redirecting for customers trying to understand the value of the integration.

mesmith027 replied:
Hi @emarsha94, 

So the type of logs we send to Datadog is the runtime logs of an application. What are you looking for in clarification on this sentence?

emarsha94 replied:
That's perfect! Can you include that and the value of sending runtime logs to Datadog? This section is to help mutual customers or new customers understand what the integration does and the value of the integration.

emarsha94 reviewed (changes requested):
Thanks for making the changes! Left some comments on the PR. Additionally, please include a dashboard for the integration. Thanks!

alai97 commented on .github/CODEOWNERS:
```diff
@@ -143,6 +143,7 @@
 /sendmail/                               @dabcoder david.bouchare@datadoghq.com
 /signl4/                                 @rons4 ron@signl4.com @DataDog/marketplace-review
 /sigsci/                                 @signalsciences info@signalsciences.com
+/shipyard/                               @shipyard support@shipyard.build 
```
```suggestion
/shipyard/                               @shipyard support@shipyard.build @DataDog/documentation
```

alai97 commented on .github/CODEOWNERS:
```diff
@@ -472,6 +473,8 @@
 /sigsci/*metadata.csv                    @signalsciences info@signalsciences.com @DataDog/documentation
 /sigsci/manifest.json                    @signalsciences info@signalsciences.com @DataDog/documentation
 /sigsci/README.md                        @signalsciences info@signalsciences.com @DataDog/documentation
+/shipyard/README.md                      @shipyard Support@shipyard.build
+/shipyard/manifest.json                  @shipyard support@shipyard.build
```
```suggestion
/shipyard/README.md                      @shipyard Support@shipyard.build @DataDog/documentation
/shipyard/manifest.json                  @shipyard support@shipyard.build @DataDog/documentation
```

anawishnoff commented:
Hi @mesmith027, echoing @emarsha94's comment that you'll need to create a dashboard for this integration. You can read about creating out of the box integration dashboards [here](https://docs.datadoghq.com/developers/integrations/create-an-integration-dashboard). Let us know if you have any questions!

mesmith027 commented:
Just a quick update for you all: The dashboard is complete I am just having our team review it and once they are happy I will export and add it to the manifest file as requested 😄 

Thanks so much for your help and patience 🙏

emarsha94 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,58 @@
+    "sales_email": "sales@shipyard.build"
+  },
+  "oauth": {},
+  assets[dashboards["Shipyard"]]: {"title":"Shipyard-Integration","description":"## Shipyard Dashboard Integration\n\nThe Shipyard dashboard allows you to see your application runtime logs in Datadog to take advantage of the many valuable features for log analysis. This example shows all the standard output runtime logs from your shipyard environment. ","widgets":[{"id":7865479500406768,"definiti ...
```
The dashboard json should be under the dashboard section

emarsha94 reviewed (changes requested):
Left a few comments on the PR! It looks like the dashboard json was added to the wrong section. This needs to be moved to the dashboard section.

emarsha94 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,58 @@
+      }
+    ],
+    "classifier_tags": [
+      "Category::Automation",
+      "Category::AWS",
+      "Category::Azure",
+      "Category::Cloud",
+      "Category::Collaboration",
+      "Category::Google Cloud",
+      "Category::Issue Tracking",
+      "Category::Orchestration",
+      "Category::Containers",
+      "Category::Developer Tools",
+      "Category::Kubernetes",
```
For context - We strongly suggest picking the top 3 categories

yzhan289 commented on .github/CODEOWNERS:
```diff
@@ -146,6 +146,9 @@
 /signl4/                                 @rons4 ron@signl4.com @DataDog/ecosystems-review
 /sigsci/                                 @signalsciences info@signalsciences.com
+/shipyard/                               @shipyard support@shipyard.build 
+/shoreline/                              @ivang-shoreline @ciprian-shoreline @alinshoreline @cioc-shoreline @cristi-shoreline support@shoreline.io @DataDog/marketplace-review
```
I think you may need to rebase from `master` since `@DataDog/marketplace-review` has changed their name to `@DataDog/ecosystems-review`.

yzhan289 commented on shipyard/assets/service_checks.json:
Note: not in this file but in `shipyard_overview.json`.

yzhan289 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,58 @@
+    ]
+  },
+  "assets": {
+    "dashboards": {},
```
Once you move the dashboard JSON content to the right file, you should link that file in this manifest.json file. Example: https://github.com/DataDog/integrations-extras/blob/master/akamai_datastream_2/manifest.json#L46

mesmith027 commented:
Hey @emarsha94 and @yzhan289, 

I addressed your issues and added int he image of the sample dashboard to the images folder. 

Thanks for your help!

mesmith027 commented:
> Hey @mesmith027 👋 , looks like validations are failing for the dashboard with the following error:
> 
> ```
> JSONDecodeError: Expecting property name enclosed in double quotes: line 49 
> ```
> 
> Could you check if the current JSON is able to be directly imported into Datadog UI? There might be a formatting error.

Hey! 

I just fixed it I believe! 

I re-exported the dashboard and added that to the file. 

Then re-tested that that it would import into a new dashboard and it seems to work as expected now.

yzhan289 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,49 @@
+    "homepage": "https://shipyard.build/",
+    "sales_email": "sales@shipyard.build"
+  },
+  "oauth": {},
```
```suggestion
  "oauth": {}
```
Turns out this was the actual cause of the failing validations.

yzhan289 commented:
Hey @mesmith027, looks like there's some issues with the manifest. I think @emarsha94 should be able to help you with those issues!

emarsha94 commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,49 @@
+  "assets": {
+    "dashboards": {
+      "Shipyard Sample Dashboard": "assets/dashboards/shipyard_overview.json"},
+    "monitors": {},
```
```suggestion
```

emarsha94 commented on shipyard/metadata.csv:
Please remove this file

emarsha94 commented on shipyard/README.md:
```diff
@@ -0,0 +1,60 @@
+
+This is useful for: 
+
+- monitoring your application’s runtime status
+- using Datadog to enrich and parse your runtime logs
+- uncover trends and compare the logs from your application’s environments using Datadog’s easy to use interface
```
```suggestion
- monitoring your application's runtime status
- using Datadog to enrich and parse your runtime logs
- uncover trends and compare the logs from your application's environments using Datadog’s easy to use interface
```

anawishnoff commented on shipyard/CHANGELOG.md:
```diff
@@ -0,0 +1,2 @@
+# CHANGELOG - Shipyard
+
```
```suggestion
## 1.0.0 / 2023-06-30

* [FEATURE] Initial Shipyard Integration Tile.
```

anawishnoff commented on shipyard/README.md:
```diff
@@ -0,0 +1,60 @@
+
+## Overview
+
+Shipyard unblocks DevOps teams from bulky, outdated production pipelines. We resolve the pre-production or staging bottleneck with our agile on-demand ephemeral (testing or preview) environments.
```
```suggestion
Shipyard unblocks DevOps teams from bulky, outdated production pipelines. We resolve pre-production or staging bottlenecks with our agile on-demand ephemeral (testing or preview) environments.
```

anawishnoff commented on shipyard/README.md:
```diff
@@ -0,0 +1,60 @@
+
+Shipyard unblocks DevOps teams from bulky, outdated production pipelines. We resolve the pre-production or staging bottleneck with our agile on-demand ephemeral (testing or preview) environments.
+
+The Datadog Shipyard integration automatically passes your [Shipyard][1] ephemeral environment logs to your Datadog dashboard. For every application you have enables the Datadog logs for, Shipyard sends the runtime logs of your application to your dashboard. 
```
```suggestion
The Datadog Shipyard integration automatically passes your [Shipyard][1] ephemeral environment logs to Datadog, for all applications on which you have enabled Datadog logs. 
```
Suggesting to shorten this text a bit, but let me know if this still makes sense (feel free to correct if not)

anawishnoff commented on shipyard/README.md:
```diff
@@ -0,0 +1,60 @@
+
+1. Navigate to the [API Keys page][5] and create an API key.
+2. In your Shipyard dashboard, go to **Settings**. 
+3. Under the Datadog section, enter your Datadog API key and the Datadog site URL. *Note: The site URL is not required and therefore we suggest you try your connection with just your API key to begin with*
```
```suggestion
3. Under the Datadog section, enter your Datadog API key and the Datadog site URL. 
```
I think it would make it simpler to either require the site URL, or just don't include the optional line above.

anawishnoff commented on shipyard/README.md:
```diff
@@ -0,0 +1,60 @@
+3. Under the Datadog section, enter your Datadog API key and the Datadog site URL. *Note: The site URL is not required and therefore we suggest you try your connection with just your API key to begin with*
+   ![datadog-input][6]
+   1. If you do not see a Datadog section here, email [support@shipyard.build][3] to enable this feature for your organization.
+   2. To get the Datadog site URL, see [Getting Started with Datadog Sites][4] and compare your dashboard's site URL to the ones in the table.
```
```suggestion
   2. To get your Datadog site URL, see [Getting Started with Datadog Sites][4].
```

anawishnoff commented on shipyard/README.md:
```diff
@@ -0,0 +1,60 @@
+   1. If you do not see a Datadog section here, email [support@shipyard.build][3] to enable this feature for your organization.
+   2. To get the Datadog site URL, see [Getting Started with Datadog Sites][4] and compare your dashboard's site URL to the ones in the table.
+3. Click **Install Datadog**.
+   1. Optionally, toggle `enable Datadog logging for all environments`, Shipyard will automatically send the logs for all your environments.
```
```suggestion
   1. Optionally, toggle `enable Datadog logging for all environments` if you would like Shipyard to automatically send logs for **all** of your environments.
```

anawishnoff commented on shipyard/README.md:
```diff
@@ -0,0 +1,60 @@
+
+## Setup
+
+These instructions assume you have a Shipyard account and application set up. If you do not, you can get started [for free][2]. For more information about setting up the Datadog integration, see the [Shipyard documentation][7]. 
```
Should this sentence:
> For more information about setting up the Datadog integration, see the [Shipyard documentation][7]. 

go at the bottom of the section?

alai97 replied:
```suggestion
These instructions assume you have a Shipyard account and application set up. If you do not, you can get started [for free][2]. 
```

anawishnoff commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,49 @@
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Send Shipyard logs to Datadog dashboard",
```
```suggestion
    "description": "Send Shipyard logs to Datadog",
```

anawishnoff commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,49 @@
+        "caption": "Datadog connection settings in your Shipyard dashboard",
+        "image_url": "images/datadog-input.png"
+      },
+      {
+        "media_type": "image",
+        "caption": "Shipyard logo in teal",
+        "image_url": "images/shipyard-teal.png"
+      }
```
```suggestion
```
The logo does not need to be added here either.  However, please add the `sample_dashboard.png` file.

anawishnoff commented on shipyard/manifest.json:
```diff
@@ -0,0 +1,49 @@
+    "dashboards": {
+      "Shipyard Sample Dashboard": "assets/dashboards/shipyard_overview.json"},
+    "monitors": {},
+    "logs": {},
+    "saved_views": {}
```
```suggestion
```

anawishnoff reviewed (changes requested):
More requested changes from product

alai97 commented on shipyard/README.md:
```diff
@@ -0,0 +1,60 @@
+   1. Select the **Notification Settings** tab. 
+   2. Toggle `Enable Datadog logging` for this environment.
+
+## Data Collected
```
```suggestion
For more information about setting up the Datadog integration, see the [Shipyard documentation][7]. 

## Data Collected
```

alai97 commented on shipyard/metadata.csv:
```diff
@@ -0,0 +1 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
```
You can remove this file I think!

anawishnoff replied:
Since this tile includes an integration but doesn't send in any metrics, this file is needed, but can be left blank

anawishnoff commented:
Closing as #2051 will replace this
